src/main/java/org/sagebionetworks/web/client/widget/table/modal/fileview/CreateTableViewWizardStep2.java

```diff
@@ -0,0 +1,326 @@
+import java.util.List;
+
+import org.sagebionetworks.repo.model.Entity;
+import org.sagebionetworks.repo.model.table.ColumnModel;
+import org.sagebionetworks.web.client.PortalGinInjector;
+import org.sagebionetworks.web.client.SynapseClientAsync;
+import org.sagebionetworks.web.client.widget.table.KeyboardNavigationHandler;
+import org.sagebionetworks.web.client.widget.table.modal.fileview.CreateTableViewWizard.TableType;
+import org.sagebionetworks.web.client.widget.table.modal.wizard.ModalPage;
+import org.sagebionetworks.web.client.widget.table.v2.schema.ColumnModelTableRow;
+import org.sagebionetworks.web.client.widget.table.v2.schema.ColumnModelTableRowEditorWidget;
+import org.sagebionetworks.web.client.widget.table.v2.schema.ColumnModelTableRowViewer;
+import org.sagebionetworks.web.client.widget.table.v2.schema.ColumnModelUtils;
+import org.sagebionetworks.web.client.widget.table.v2.schema.ColumnModelsView;
+import org.sagebionetworks.web.client.widget.table.v2.schema.ColumnModelsView.ViewType;
+import org.sagebionetworks.web.client.widget.table.v2.schema.ColumnModelsWidget;
+import org.sagebionetworks.web.client.widget.table.v2.schema.ColumnModelsWidget.SelectedRow;
+
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+
+/**
+ * Wizard page to edit column schema
+ * 
+ * @author Jay
+ *
+ */
+public class CreateTableViewWizardStep2 implements ColumnModelsView.Presenter, ColumnModelTableRow.SelectionPresenter, ModalPage, IsWidget {
+	ColumnModelsView editor;
+	PortalGinInjector ginInjector;
+	SynapseClientAsync synapseClient;
+	String tableId;
+	List<ColumnModel> startingModels;
+	List<ColumnModelTableRow> editorRows;
+	KeyboardNavigationHandler keyboardNavigationHandler;
+	ModalPresenter presenter;
+	// the TableEntity or View
+	Entity entity;
+	TableType tableType;
+	
+	/*
+	 * Set to true to indicate that change selections are in progress.  This allows selection change events to be ignored during this period.
+	 */
+	boolean changingSelection = false;
+	/**
+	 * New presenter with its view.
+	 * @param view
+	 */
+	@Inject
+	public CreateTableViewWizardStep2(PortalGinInjector ginInjector, SynapseClientAsync synapseClient){
+		this.ginInjector = ginInjector;
+		this.editor = ginInjector.createNewColumnModelsView();
+		this.editor.setPresenter(this);
+		this.editor.setAddAllAnnotationsButtonVisible(true);
+		this.editor.setAddDefaultFileColumnsButtonVisible(true);
+		this.synapseClient = synapseClient;
+		this.editorRows = new LinkedList<ColumnModelTableRow>();
+		keyboardNavigationHandler = ginInjector.createKeyboardNavigationHandler();
+	}
+
+	public void configure(Entity entity, TableType tableType) {
+		configure(entity, tableType, new ArrayList<ColumnModel>());
+	}
+	
+	public void configure(Entity entity, TableType tableType, List<ColumnModel> startingModels) {
+		this.changingSelection = false;
```

Comment:
set visibility of annotation buttons based on tableType (default hidden in constructor).